contracts/Builders/RefundBundleBuilder.sol

```diff
@@ -74,16 +70,24 @@ contract RefundBundleBuilder is ERC721Holder {
     }
 
     function _createRefundProvider(
-        uint256 collateralPoolId,
-        address token,
-        uint256 tokenAmount,
-        uint256 mainCoinAmount
-    ) internal returns (uint256 poolId) {
-        uint256[] memory params = new uint256[](2);
-        params[0] = collateralPoolId;
-        params[1] = mainCoinAmount.calcRate(tokenAmount);
+        address[] calldata addressParams,
+        uint256[][] calldata params
+    ) internal returns (uint256 poolId, uint256 tokenAmount) {
+        for (uint256 i = 1; i < params.length; i++) {
+            tokenAmount += params[i][0];
+        }
+        address token = addressParams[0];
+        address mainCoin = addressParams[1];
+        uint256 mainCoinAmount = params[0][0];
+        uint256[] memory collateralParams = params[0];
```

Comment:
Will be `Array accessed at an out-of-bounds or negative index` error. `CollateralProvider` must know the `RefundProvider pool ID` for token `copyVaultId`. We can't send it before it's created.

 We have a situation where the `RefundProvider` needs to know the `Collateral` poolId and the `Collateral` needs to know the `Refund` poolId. And we can't send empty pool IDs.

Reply:
We can first create `RefundProvider`, then `BundleProvider`, then `Collateral`, and then register the `Refund`.